gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -146,26 +148,12 @@ public Orchestrator(Config config, Optional<TopologyCatalog> topologyCatalog, Op
         ServiceConfigKeys.DEFAULT_FLOW_CONCURRENCY_ALLOWED);
   }
 
-  public Orchestrator(Config config, Optional<TopologyCatalog> topologyCatalog, Optional<DagManager> dagManager, Optional<Logger> log) {
-    this(config, topologyCatalog, dagManager, log, true);
-  }
-
-  public Orchestrator(Config config, Optional<TopologyCatalog> topologyCatalog, Optional<DagManager> dagManager, Logger log) {
-    this(config, topologyCatalog, dagManager, Optional.of(log));
+  @Inject
+  public Orchestrator(Config config, FlowStatusGenerator flowStatusGenerator, Optional<TopologyCatalog> topologyCatalog,
+      Optional<DagManager> dagManager, Optional<Logger> log) {
+    this(config, topologyCatalog, dagManager, log, flowStatusGenerator, true);
```

Comment:
Are we sure about removing these public constructors? Looks redundant to me though, but can be used by external users.

Reply:
This class should be created using dependency injection, so other constructors should not be needed.